machine/v1beta1/types_azureprovider.go

```diff
@@ -215,6 +223,48 @@ type OSDisk struct {
 	CachingType string `json:"cachingType,omitempty"`
 }
 
+// DataDisk specifies the parameters that are used to add one or more data disks to the machine.
+type DataDisk struct {
+	// NameSuffix is the suffix to be appended to the machine name to generate the disk name.
+	// Each disk name will be in format <machineName>_<nameSuffix>.
+	// NameSuffix name must start and finish with an alphanumeric character and can only contain letters, numbers, underscores, periods or hyphens.
+	// The overall disk name must not exceed 80 chars in length.
+	// +kubebuilder:validation:Pattern:=`^[a-zA-Z0-9](?:[\w\.-]*[a-zA-Z0-9])?$`
+	// +kubebuilder:validation:MaxLength:=78
+	// +kubebuilder:validation:Required
+	NameSuffix string `json:"nameSuffix"`
+	// DiskSizeGB is the size in GB to assign to the data disk.
+	// +kubebuilder:validation:Minimum=4
+	// +kubebuilder:validation:Required
+	DiskSizeGB int32 `json:"diskSizeGB"`
```

Comment:
Why an int32 instead of a resource.Quantity?

Reply:
For a couple of reasons.
- This is the same of what is already set in OS Disk
- The [underlying Machine type provided](https://github.com/openshift/machine-api-provider-azure/blob/02da4328b3615ad84a2b808e203686fa4182d27d/pkg/cloud/azure/services/virtualmachines/virtualmachines.go#L263) by the cloud provider SDK accepts this as an `int32` on a field named `DiskSizeGB`
- This cannot be set at a finer grained level in the UI, they enforce it to be a Gigabyte integer
- In designing this, I tried to match the upstream API interface as close as possible to allow easier feature migration between MAPI and CAPI, and this is how they are implementing it

Reply:
> This is the same of what is already set in OS Disk

sins of the past don't rule the future.

> The underlying Machine type provided by the cloud provider SDK accepts this as an int32 on a field named DiskSizeGB

This API follows kube conventions.  The choices of the underlying API don't matter.

> This cannot be set at a finer grained level in the UI, they enforce it to be a Gigabyte integer

This one is slightly more compelling, but I think the argument is stronger for, "the additional flexibility of resource.Quantity doesn't provide value to the user here"

> In designing this, I tried to match the upstream API interface as close as possible to allow easier feature migration between MAPI and CAPI, and this is how they are implementing it

This isn't hugely compelling.  A CAPI api mistake doesn't need to be reflected in our API.

Reply:
> the additional flexibility of resource.Quantity doesn't provide value to the user here

Yes that's probably the best way to put that :)

So, to come to a verdict, does it sound reasonable to keep the `int32` to you? Thanks

Reply:
Yes, I find that reasoning good.

Reply:
Yes, I find that reasoning good.